packages/server/src/modules/user/seed-users-with-books.command.ts

```diff
@@ -0,0 +1,518 @@
+    description:
+      "The amount of displacement of the random number of copies between every generated user\nAccepted values: natural numbers (default: 5)",
+  })
+  parseRandomDisplacement(val?: string) {
+    if (
+      val !== undefined &&
+      (parseInt(val) <= 0 ||
+        parseInt(val) !== parseFloat(val) ||
+        isNaN(parseInt(val)))
+    ) {
+      throw new Error(
+        "Invalid displacement, must be an integer greater than zero",
+      );
+    }
+    return val !== undefined ? parseInt(val) : val;
+  }
+
+  @Option({
+    flags: "-c, --clear",
+    description:
+      "Makes the command stop after clearing the previous seeded data",
+  })
+  parseNothing(_?: string) {
+    return true;
+  }
+
+  async run(
+    _: string[],
+    options?: SeedUsersWithBooksCommandOptions,
+  ): Promise<void> {
+    // TODO: make the clear of the seeded users optional once seeding upsert is allowed
+    try {
+      await this.#clearRetailLocationSeededUsers();
+    } catch (error) {
+      this.logger.error(
+        "There was a problem while cleaning any previously seeded users",
+        error,
+      );
+      return;
+    }
+
+    this.logger.log("Seed data erased successfully");
+
+    if (options?.clear) {
+      // Option to only clear the current seeded data
+      return;
+    }
+
+    this.logger.verbose(
+      `Range of copies created per user: ${this.BOOKS_PER_USER_BASE} - ${this.BOOKS_PER_USER_BASE + (options?.random ?? this.DEFAULT_RANDOM_DISPLACEMENT)}`,
+    );
+
+    for (const id of ["re", "mo"]) {
+      await this.#seedRetailLocation(id, options?.random);
+    }
+
+    this.logger.log("Seeding complete");
+  }
+
+  async #seedRetailLocation(
+    retailLocationId: string,
+    randomDisplacement = this.DEFAULT_RANDOM_DISPLACEMENT,
+  ) {
+    let retailLocation: RetailLocation | undefined;
+    try {
+      retailLocation = await this.prisma.retailLocation.findUniqueOrThrow({
+        where: {
+          id: retailLocationId,
+        },
+      });
+    } catch {
+      this.logger.error(
+        "Something went wrong while fetching the current location",
+      );
+      return;
+    }
+
+    this.logger.log(`Creating the users for ${retailLocation.name}`);
+
+    // TODO: modify this to make seeding upsert possible
+    for (let i = 0; i < this.USERS_AMOUNT; i++) {
+      try {
+        await this.#createBuyerAndSeller(i, randomDisplacement, retailLocation);
+      } catch (error) {
+        this.logger.error(error);
+        return;
+      }
+    }
+
+    this.logger.log(`Users for ${retailLocation.name} created successfully`);
+  }
+
+  async #createBuyerAndSeller(
+    index: number,
+    randomDisplacement: number,
+    retailLocation: RetailLocation,
+  ) {
+    try {
+      const bookIds = (
+        await this.prisma.book.findMany({
+          take: randomInt(
+            this.BOOKS_PER_USER_BASE,
+            this.BOOKS_PER_USER_BASE + randomDisplacement,
+          ),
```

Comment:
In reality since there's already at least 3k books inside each retail location it wouldn't really be a problem; the only real problem comes when the books aren't present in the db to begin with; if there is really a need I'd rather add a check to see if the generated random number is larger than the amount of books for the retail location and potentially regenerate until it is small enough (still I doubt someone is going to input 4000 as the random displacement).

Reply:
> In reality since there's already at least 3k books inside each retail location it wouldn't really be a problem

Yes exactly I agree with this, that's why I said this was just a suggestion.

---

But anyway, if I have to think about a solution here, you can always check that `displacement` is > 0 and < numberOfBooks of targeted retail location.
If the `displacement` surpasses these values, we simply throw an error to the user and that's it :)